drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -0,0 +1,1829 @@
+		}
+
+		LOG_DBG("DHCP Server started successfully");
+		s_nxp_wifi_UapActivated = true;
+		break;
+	case WLAN_REASON_UAP_CLIENT_ASSOC:
+		LOG_DBG("WLAN: UAP a Client Associated");
+		LOG_DBG("Client => ");
+		print_mac((const char *)data);
+		LOG_DBG("Associated with Soft AP");
+		break;
+	case WLAN_REASON_UAP_CLIENT_CONN:
+		wlan_get_current_uap_network(&nxp_wlan_network);
+#ifdef CONFIG_NXP_WIFI_11AX
+			if (nxp_wlan_network.dot11ax) {
+				ap_sta_info.link_mode = WIFI_6;
+			}
+#endif
+#ifdef CONFIG_NXP_WIFI_11AC
+			else if (nxp_wlan_network.dot11ac) {
+				ap_sta_info.link_mode = WIFI_5;
+			}
+#endif
+			else if (nxp_wlan_network.dot11n) {
+				ap_sta_info.link_mode = WIFI_4;
+			} else {
+				ap_sta_info.link_mode = WIFI_3;
+			}
+
+		memcpy(ap_sta_info.mac, data, WIFI_MAC_ADDR_LEN);
+		ap_sta_info.mac_length  = WIFI_MAC_ADDR_LEN;
+		ap_sta_info.twt_capable = status.twt_capable;
+
+		wifi_mgmt_raise_ap_sta_connected_event(g_uap.netif, &ap_sta_info);
+		LOG_DBG("WLAN: UAP a Client Connected");
+		LOG_DBG("Client => ");
+		print_mac((const char *)data);
+		LOG_DBG("Connected with Soft AP");
+		break;
+	case WLAN_REASON_UAP_CLIENT_DISSOC:
+		memcpy(ap_sta_info.mac, disassoc_resp->sta_addr, WIFI_MAC_ADDR_LEN);
+		wifi_mgmt_raise_ap_sta_disconnected_event(g_uap.netif, &ap_sta_info);
+
+		LOG_DBG("WLAN: UAP a Client Dissociated:");
+		LOG_DBG(" Client MAC => ");
+		print_mac((const char *)(disassoc_resp->sta_addr));
+		LOG_DBG(" Reason code => ");
+		LOG_DBG("%d", disassoc_resp->reason_code);
+		break;
+	case WLAN_REASON_UAP_STOPPED:
+		net_eth_carrier_off(g_uap.netif);
+		LOG_DBG("WLAN: UAP Stopped");
+
+		net_dhcpv4_server_stop(g_uap.netif);
+		LOG_DBG("DHCP Server stopped successfully");
+		s_nxp_wifi_UapActivated = false;
+		break;
+#endif
+	case WLAN_REASON_PS_ENTER:
+		LOG_DBG("WLAN: PS_ENTER");
+		break;
+	case WLAN_REASON_PS_EXIT:
+		LOG_DBG("WLAN: PS EXIT");
+		break;
+#ifdef CONFIG_NXP_WIFI_SUBSCRIBE_EVENT_SUPPORT
+	case WLAN_REASON_RSSI_HIGH:
+	case WLAN_REASON_SNR_LOW:
+	case WLAN_REASON_SNR_HIGH:
+	case WLAN_REASON_MAX_FAIL:
+	case WLAN_REASON_BEACON_MISSED:
+	case WLAN_REASON_DATA_RSSI_LOW:
+	case WLAN_REASON_DATA_RSSI_HIGH:
+	case WLAN_REASON_DATA_SNR_LOW:
+	case WLAN_REASON_DATA_SNR_HIGH:
+	case WLAN_REASON_LINK_QUALITY:
+	case WLAN_REASON_PRE_BEACON_LOST:
+		break;
+#endif
+	default:
+		LOG_WRN("WLAN: Unknown Event: %d", reason);
+	}
+	return 0;
+}
+
+static int nxp_wifi_wlan_init(void)
+{
+	int status = NXP_WIFI_RET_SUCCESS;
+	int ret;
+
+	if (s_nxp_wifi_State != NXP_WIFI_NOT_INITIALIZED) {
+		status = NXP_WIFI_RET_FAIL;
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		k_event_init(&s_nxp_wifi_SyncEvent);
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		ret = wlan_init(wlan_fw_bin, wlan_fw_bin_len);
+		if (ret != WM_SUCCESS) {
+			status = NXP_WIFI_RET_FAIL;
+		}
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+		s_nxp_wifi_State = NXP_WIFI_INITIALIZED;
+	}
+
+	if (status != NXP_WIFI_RET_SUCCESS) {
+		return -1;
+	}
```

Comment:
Same thing here, by ordering the check differently one check can be omitted

Reply:
thanks for your advice, updated